proposals/lip-0053.md

```diff
@@ -252,178 +479,192 @@ crossChainUpdateTransactionParams = {
-)
-```
-
-If `params` contains a non-empty `activeValidatorsUpdate` property (with deserialized value not equal to `{activeValidatorsUpdate:[]}`), or a non-zero `newCertificateThreshold`, `params` is valid if:
-
-* `params` contains a non-empty `certificate`.
-* `params.activeValidatorsUpdate` has the correct format:
-  * for all `activeValidator` in `params.activeValidatorsUpdate` we have that `activeValidator.blsKey` is a BLS public key, hence it is 48 bytes long.
-  * every `activeValidator` in `params.activeValidatorsUpdate` has a different `activeValidator.blsKey`.
-  * `params.activeValidatorsUpdate` is in lexicographical order of `blsKey`.
-* `certificate.validatorsHash == bft.computeValidatorsHash(newActiveValidators, newCertificateThreshold)` where
-  * `newActiveValidators` is obtained as `partnerValidators.activeValidators` updated with `params.activeValidatorsUpdate`, see ["Update Active Validators" section](#update-active-validators) below.
-  * `newCertificateThreshold` is `params.newCertificateThreshold` if it is non-zero, `partnerValidators.certificateThreshold` otherwise.
-
-#### InboxUpdate Validity
-
-If `params` contains a non-empty `certificate` and an `inboxUpdate`, the validity of the latter is checked by:
-
-* Let `newInboxRoot`, `newInboxAppendPath` and `newInboxSize` be the resulting new root, append path and size of the Merkle tree that would be obtained if all messages from `crossChainMessages` would be appended to the `partnerChannel` inbox. Note that `partnerChannel.inbox` is not updated here.
-* If `inboxUpdate` contains a non-empty `messageWitness`, then update `newInboxRoot` to the output of `calculateRootFromRightWitness(newInboxSize, newInboxAppendPath, inboxUpdate.messageWitness.siblingHashes)` as specified in [LIP 0031](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0031.md).
-* Then validate the newly updated root against the certificate state root using the provided `outboxRootWitness`. Using notation from [LIP 0039][lip-0039], this is done via the function `verify(queryKeys, proof, certificate.stateRoot)` with:
-
-```python
-queryKeys = [outboxKey],
-
-proof = {
-    siblingHashes: outboxRootWitness.siblingHashes,
-    queries: [{
-        key: outboxKey,
-        value: newInboxRoot,
-        bitmap: outboxRootWitness.bitmap
-    }],
-}
-
-outboxKey = MODULE_ID_INTEROPERABILITY
-            || STORE_PREFIX_OUTBOX
-            || SHA-256(uint32be(partnerChainID)).
+def execute(trs: Transaction) -> None:
+    # Verify certificate signature. We do it here because if it fails, the transaction fails rather than being invalid.
+    verifyCertificateSignature(trs)
+    # Process cross-chain messages in inbox update.
+    crossChainMessages = trs.params.inboxUpdate.crossChainMessages
+
+    # First process basic checks for all CCMs.
+    for ccmBytes in crossChainMessages:
+        # Verify general format. Past this point, we can access ccm root properties.
+        try:
+            validateFormat(ccm)
```

Comment:
`ccm` is not defined and `validateFormat` assumes a CCM that is decoded.

Reply:
added decoding